app/src/lib/stores/app-store.ts

```diff
@@ -7136,6 +7138,121 @@ export class AppStore extends TypedBaseStore<IAppState> {
       numberOfComments,
     })
   }
+
+  public async _startPullRequest(repository: Repository) {
+    const { branchesState } = this.repositoryStateCache.get(repository)
+    const { defaultBranch, tip } = branchesState
+
+    if (defaultBranch === null || tip.kind !== TipState.Valid) {
+      return
+    }
+
+    const currentBranch = tip.branch
+    const gitStore = this.gitStoreCache.get(repository)
+
+    const pullRequestCommits = await gitStore.getCommitsBetweenBranches(
+      defaultBranch,
+      currentBranch
+    )
```

Comment:
We should probably check if there are no commits at all, right?
```suggestion
    const pullRequestCommits = await gitStore.getCommitsBetweenBranches(
      defaultBranch,
      currentBranch
    )
    
    if (pullRequestCommits.length === 0) {
      return
    }
```

Reply:
Very good point that I intentionally left out at the moment,  because I didn't know if we would like to do something like:
![image](https://user-images.githubusercontent.com/75402236/190196387-f2ad5166-26f6-4409-a379-f05027fa4f89.png)

I think we will want this because we are going to let a user change the base branch in the modal and we wouldn't want it to just close the dialog. :D However, I think I may refactor the initialization logic a little to allow a pr state without a change set. I was planning to do that in a later PR after discussion with group on possibility of a no changes dialog. But now that I think on it more, I may want to just go ahead and plan for it.

Reply:
Added an empty change set for this case.

Reply:
Yeah, your change makes much more sense then. I was concerned about the `commitSHAs[0]` which would be `undefined` in that case, but now it's all good 🙌